src/ewatercycle/plugins/wflow/model.py

```diff
@@ -85,141 +58,79 @@ def _setup_default_config(self):
+        return values
+
+    # TODO: create setup with custom docstring to explain extra kwargs
+    # (start_time, end_time)
+
+    def _make_cfg_file(self, **kwargs) -> Path:
+        """Create a new wflow config file and return its path."""
         if "start_time" in kwargs:
-            cfg.set("run", "starttime", _iso_to_wflow(kwargs["start_time"]))
+            self._config.set("run", "starttime", _iso_to_wflow(kwargs["start_time"]))
         if "end_time" in kwargs:
-            cfg.set("run", "endtime", _iso_to_wflow(kwargs["end_time"]))
-
-        updated_cfg_file = to_absolute_path(
-            "wflow_ewatercycle.ini", parent=self.work_dir
-        )
-        with updated_cfg_file.open("w") as filename:
-            cfg.write(filename)
-
-        self.bmi = start_container(
-            image_engine=_version_images[self.version],
-            work_dir=self.work_dir,
-            timeout=300,
-        )
-
-        return (
-            str(updated_cfg_file),
-            str(self.work_dir),
-        )
-
-    def _setup_working_directory(self, cfg_dir: Optional[str] = None):
+            self._config.set("run", "endtime", _iso_to_wflow(kwargs["end_time"]))
+
+        cfg_file = to_absolute_path("wflow_ewatercycle.ini", parent=self._work_dir)
+        with cfg_file.open("w") as filename:
+            self._config.write(filename)
+
+        return cfg_file
+
+    def _make_cfg_dir(self, cfg_dir: Optional[str] = None, **kwargs) -> Path:
+        """Create working directory for parameter sets, forcing and wflow config."""
         if cfg_dir:
-            self.work_dir = to_absolute_path(cfg_dir)
+            self._work_dir = to_absolute_path(cfg_dir)
         else:
             timestamp = datetime.datetime.now(datetime.timezone.utc).strftime(
                 "%Y%m%d_%H%M%S"
             )
-            self.work_dir = to_absolute_path(
+            self._work_dir = to_absolute_path(
                 f"wflow_{timestamp}", parent=CFG.output_dir
             )
         # Make sure parents exist
-        self.work_dir.parent.mkdir(parents=True, exist_ok=True)
+        self._work_dir.parent.mkdir(parents=True, exist_ok=True)
 
         assert self.parameter_set
-        shutil.copytree(src=self.parameter_set.directory, dst=self.work_dir)
+        shutil.copytree(src=self.parameter_set.directory, dst=self._work_dir)
         if self.forcing:
             forcing_path = to_absolute_path(
                 self.forcing.netcdfinput, parent=self.forcing.directory
             )
-            shutil.copy(src=forcing_path, dst=self.work_dir)
+            shutil.copy(src=forcing_path, dst=self._work_dir)
 
-    def _coords_to_indices(
-        self, name: str, lat: Iterable[float], lon: Iterable[float]
-    ) -> Iterable[int]:
-        """Convert lat/lon values to index.
+        return self._work_dir
 
-        Args:
-            lat: Latitudinal value
-            lon: Longitudinal value
+    def get_parameters(self):
+        self._post_init()
```

Comment:
this post_init is not defined, should it call `_parse_config` instead?

Reply:
Oh yeah you're right. I guess the tests didn't catch any of that... I think renaming the `_parse_config` is best. 

Doing these operations in a `post_init` in all models would make them look more similar.